code/datums/uplink/uplink_categories.dm

```diff
@@ -1,49 +1,117 @@
 /datum/uplink_category
 	var/name = ""
 	var/list/datum/uplink_item/items
+	var/list/datum/antagonist/antag_roles  // added this so category wide restrictions can be placed to make it eaiser to set restrictions
 
 /datum/uplink_category/New()
 	..()
 	items = list()
+	antag_roles = list()
```

Comment:
This is a dumb way to do it. Even though the way it generates items is also dumb, it shouldn't be perpetuated. You can just assign `var/list/datum/antagonist/antag_roles = list()` in the prototype. Same for all of the subclasses, there's no need to write out the New() proc for them.

Unless there's an empirical need. Though, I don't see any.